acasclient/experiment.py

```diff
@@ -36,10 +37,35 @@ def fetch_analysis_groups_if_missing(func):
     """
     @wraps(func)
     def wrapper(self, *args, **kwargs):
+        # Check if 'client' is in kwargs
+        aclient = kwargs.get('client', None)
+        
+        # If 'client' is not in kwargs, check if it's in args
+        if aclient is None:
+            for arg in args:
+                if isinstance(arg, client):
+                    aclient = arg
+                    break
+        
+        if aclient is None:
+            if self.client is None:
+                raise Exception("No client provided and no client found in the experiment")
+            aclient = self.client
+        
         if ANALYSIS_GROUPS_KEY not in self:
-            experiment_dict = self.client.get_experiment_by_code(self.code, full=True)
-            self.__init__(experiment_dict, client=self.client)
-        return func(self)
+            experiment_dict = aclient.get_experiment_by_code(self.code, full=True)
+            self[ANALYSIS_GROUPS_KEY] = experiment_dict[ANALYSIS_GROUPS_KEY]
+        return func(self, *args, **kwargs)
+    return wrapper
+
+def fetch_analysis_groups_if_images_expertiment(func):
```

Comment:
typo in `expertiment`, should be `experiment`

Reply:
Thanks fixed.